### PR TITLE
[US435530]: Persist changes made to release dockerized version of UAA 4.30.0

### PR DESCRIPTION
**Pull request by emarmol**
Story: [US435530](https://rally1.rallydev.com/#/detail/userstory/410715018988/tasks?fdp=true): Persist changes made to release dockerized version of UAA 4.30.0 for future release
Task: Persist changes in UAA docker image name and tag in predix_extension_4.30.0 branch

Successful build unit & integration tests: https://i.ci.build.ge.com/bjmjek1k/ci/job/UAA/job/Build%20n%20Test/view/change-requests/job/PR-152/11/

### CODE REVIEW

File: DockerizeJenkinsfile [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
     agent {
         docker {
-            image 'registry.gear.ge.com/dig-predix-security-services/uaa-ci-testing-with-docker:latest'
+            image 'registry.gear.ge.com/dig-predix-security-services/uaa-ci-testing:latest'
             label 'dind'
             args '-v /var/lib/docker/.gradle:/root/.gradle -v /var/run/docker.sock:/var/run/docker.sock --add-host "localhost":127.0.0.1'
         }
     }
     environment {
+        USE_RELEASE_ARTIFACTS = true
```
Comment (irinaepshteyn) [inline DockerizeJenkinsfile:16]: Release artifacts should be used only when on release branch.  Use use_release_artifacts() to assign.
It was temporary forced to true on release_4.30.0_dockerized branch.

File: docker.gradle [hunk 2]
```diff
@@ -6,6 +6,9 @@ DOCKER_REPO_USER
 DOCKER_REPO_USER_PW
 DOCKER_CONTAINER_NAME
 
+To build UAA docker image using release artifacts:
+export USE_RELEASE_ARTIFACTS=true
```
Comment (irinaepshteyn) [inline docker.gradle:10]: Should read from USE_RELEASE_ARTIFACTS set in DockerizeJenkinsfile

File: uaa/src/test/resources/sso_metadata.xml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+<md:EntityDescriptor ID="y0f_8PtWxWnNp3rlzZT4-VGoQbY" cacheDuration="PT1440M" entityID="gefssstg" xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata"><ds:Signature xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
```
Comment (irinaepshteyn) [inline uaa/src/test/resources/sso_metadata.xml:1]: Please revert this change. This change did not fix SAML acceptance tests and SamlLoginAT is excluded for now.  There is a separate story to fix SAML acceptance tests.  We will update metadata then.

File: DockerizeJenkinsfile [hunk 4]
```diff
@@ -80,6 +79,44 @@ pipeline {
...
+                }
+            }
+        }
+        stage('Push Docker Image') {
+            environment {
+                DOCKER_USER = credentials('dtr-docker-repo-pusher')
+            }
+            // Publish the UAA docker image only if we are on release branch
+            when {
+                branch pattern: 'release_\\d+\\.\\d+\\.\\d+_dockerized', comparator: 'REGEXP'
```
Comment (irinaepshteyn) [inline DockerizeJenkinsfile:110]: Remove "_dockerized"; it was added temporarily to publish UAA docker image from release_4.30.0_dockerized branch.

File: docker.gradle (hunk 2 above)
Comment (emarmol) [reply docker.gradle:10]: This comment seems like dev information for when we want to require docker image to use release artifacts. [Line 48](https://github.com/GESoftware-CF/uaa/pull/152/files/6a65e5daa72778311f706c4b7ea5a6ca7cdab047#diff-b4d7dd8622610fcebe4ad687a457cbb7R48) is the actual implementation which does read from System env set in DockerizeJenkinsfile. Do you want me to remove the comment or update the wording?

Comment (irinaepshteyn) [reply docker.gradle:10]: Oops... Sorry, didn't realize it's commented out.  Never mind.  Doesn't worth changing.

File: DockerizeJenkinsfile (hunk 1 above)
Comment (emarmol) [reply DockerizeJenkinsfile]: Fixed in last commit

File: uaa/src/test/resources/sso_metadata.xml (hunk 3 above)
Comment (emarmol) [reply uaa/src/test/resources/sso_metadata.xml]: Removed

File: DockerizeJenkinsfile (hunk 4 above)
Comment (emarmol) [reply DockerizeJenkinsfile]: Removed

**Comment by irinaepshteyn** [comment]
@emarmol, approved.  Go ahead and merge
